### PR TITLE
Fs cleaning

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
The travis error is checkpatch complaining over function pointers as members of a struct as usual.

**Comment by jforissier** [comment]
@jenswi-linaro yes and this one is not so easy to fix, the root cause is checkpatch.pl does not know our types (TEE_Result etc.). I have a quick and dirty patch for checkpatch, to make it look for custom typedefs in a local file, but I'd like to clean it up a bit before sending it to the LKML.

**Comment by jenswi-linaro** [comment]
@jforissier OK, no worries.

**Comment by jforissier** [comment]
Nice cleanup.
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey, GP)`

File: core/tee/tee_rpmb_fs.c [hunk 1]
```diff
@@ -26,21 +26,24 @@
...
 #include <kernel/tee_common_otp.h>
+#include <kernel/tee_misc.h>
 #include <kernel/thread.h>
 #include <mm/core_memprot.h>
 #include <mm/tee_mm.h>
 #include <optee_msg_supplicant.h>
 #include <stdlib.h>
-#include <string.h>
 #include <string_ext.h>
+#include <string.h>
```
Comment (jenswi-linaro) [inline core/tee/tee_rpmb_fs.c]: I'm sorting with `sort`.
Actually I'm selecting the lines in vim and and then pipe it to sort with `!sort`.

**Comment by jenswi-linaro** [comment]
Tags applied, seek reference in `secure_storage_sql.md` removed also.

File: core/tee/tee_rpmb_fs.c (hunk 1 above)
Comment (etienne-lms) [inline core/tee/tee_rpmb_fs.c]: ok, thanks for the info. I tried it and it gives strange sorting results (at least not want I would expect) when using non alphanum characters. 
```
foo/my_ack.h
foo/my/ack.h
foo/mystery.h
foo/my_string.h
foo/my/string.h
```
nevertheless less, that's quite minor stuff.

Comment (jforissier) [inline core/tee/tee_rpmb_fs.c]: I'd be tempted to consider `/` and `.` as separators, for a first level of sorting, then alphanumerical sort for each component. But even on the LKML there is no clear rule as to which order is actually the good one.

I'm fine with the default `sort` algorithm.

Comment (etienne-lms) [inline core/tee/tee_rpmb_fs.c]: I guess the main idea is that order of the include should not be important so alpha-sorting is a good compromise for easy maintenance of source files. Let's not be to picky on this (especially concerning my pull-requests :)

File: core/tee/tee_fs_rpc.c [hunk 2]
```diff
@@ -27,12 +27,15 @@
 
 #include <assert.h>
 #include <kernel/thread.h>
+#include <kernel/tee_misc.h>
```
Comment (etienne-lms) [inline core/tee/tee_fs_rpc.c]: major: shift <kernel/tee_misc.h> above <kernel/thread.h>
and shift <tee/tee_svc_storage.h> below <tee/tee_pobj.h> at line 37.

**Comment by jenswi-linaro** [comment]
If you're still OK with the small change I squash and rebase

**Comment by jforissier** [comment]
Fine with me.

**Comment by jenswi-linaro** [comment]
Squashed and rebased

**Pull request merged**